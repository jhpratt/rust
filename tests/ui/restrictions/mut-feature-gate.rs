// gate-test-mut_restriction
// compile-flags: --crate-type=lib
// revisions: with_gate without_gate
//[with_gate] check-pass

#![cfg_attr(with_gate, feature(mut_restriction))]

pub struct Foo {
    pub mut(self) alpha: u8, //[without_gate]~ ERROR
}
