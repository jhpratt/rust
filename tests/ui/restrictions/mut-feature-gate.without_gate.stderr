error[E0658]: `mut` restrictions are experimental
  --> $DIR/mut-feature-gate.rs:9:9
   |
LL |     pub mut(self) alpha: u8,
   |         ^^^^^^^^^
   |
   = note: see issue #105077 <https://github.com/rust-lang/rust/issues/105077> for more information
   = help: add `#![feature(mut_restriction)]` to the crate attributes to enable

error: aborting due to previous error

For more information about this error, try `rustc --explain E0658`.
